### DISCUSSION TITLE
cmake cask does not install cmake commands in /usr/local/opt

### DISCUSSION THREAD

**Discussion by rfritz** [Homebrew/discussions#571] [Casks]
The cmake cask doesn't add the cmake CLI tools to /usr/local/opt, though it does add them to /usr/local/bin. This, in turn, keeps 'depends_on "cmake" => :build' from finding cmake if it is installed from the cmake cask.

Is this how the cask is supposed to work? Do I have a bad installation? Or was this an oversight?

**Comment by SMillerDev** [comment]
Casks do whatever upstream defined them to do. If you want to have software interact with homebrew you'll need to use a formula.

**Comment by carlocab** [comment]
`depends_on "cmake"` always means that it depends on the CMake formula. I suppose you could try to use the CMake cask to fake a `/usr/local/opt/cmake` and then do `brew install --ignore-dependencies`, but that is not recommended. I'm not certain it'll work either, even if you fake the directory correctly.

**Comment by rfritz** [comment]
The specific error message is:

> RuntimeError: /usr/local/opt/cmake not present or broken
> Please reinstall cmake. Sorry :(


I rather suspect that if /usr/local/opt/cmake was provided by the cask,
'depends_on "cmake" : build' would work.
-- 
Randolph M. Fritz || rmfritz3@gmail.com

**Reply by carlocab** [reply]
That sounds to me like `brew` is letting you do something it's actually designed not to do. In any case, as Sean has already pointed out, Homebrew has almost no control over what Casks install and where they install them. You'll need to take this up with the CMake developers.

**Reply by trg818** [reply]
Has there been any progress on this issue? I'm trying to figure out how to brew the music player Clementine (see issue #1093) and am running into the same problem with the cmake cask. Is it not possible to provide a "real" cmake brew?

**Reply by gromgit** [reply]
Have you tried compiling it with the `cmake` _formula_ instead of the cask?

**Reply by trg818** [reply]
Ok, I reinstalled the cask, then
`brew install --formula cmake`
and then
`brew link --overwrite cmake`
as suggested by the message on stdout, and that worked.
Thanks for all replies.
